### DISCUSSION TITLE
Workspace Isolation

### DISCUSSION THREAD

**Discussion by crimsonvspurple** [Kong/insomnia#3632] [Ideas]
I'd like to propose a feature set geared towards multiple separate projects. A typical workflow for that (on Windows) is using Windows Virtual Desktop to isolate multiple working environment (including browser, IDE, terminal etc). I find it particularly difficult to fit insomnia in that flow.

Here are the things insomnia needs for that:
- Ability to open a workspace directly on a new window (and optionally disable workspace switcher)
This can be utilized with command line parameters.
Bonus: option to add workspace to start menu/desktop/taskbar automatically
- Option to open each workspace (from workspace switcher) in a different window
- Upon restarting, insomnia should be able to restore all the workspaces opened (on separate windows if that's how it was before)

Simply put, make it work like Firefox where if you have multiple windows and shutdown/restart computer, it opens all those windows with specific tabs (in this case workspace).

Bonus: restore the windows to their last known Virtual Desktop (if that desktop is available).

**Comment by nijikokun** [comment]
This has come up in the past in the form of tabs, and internally on the team we have had discussions around multiple window support as well. Over the next few releases, I feel that this will become more of a discussion focus point.

**Comment by stale** [comment]
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.
